### PR TITLE
WordCount sans tests.

### PR DESCRIPTION
**Pull request by tavizon**
Working api for counting words on a webpage.

---
## Impacted Areas in Application

- None

## Deploy Notes

- This PR **will not** be deployed to a preview environment prior to QA

## Manual Testing

- [ ] `pm2 start app.js`
- [ ] `http "http://localhost:5959/ping" \\ PONG`
- [ ] `http "http://localhost:5959/api/words?url=http://www.kink.com/shoot/39563" \\ {"words":<count>}`
- [ ] `http "http://localhost:5959/api/words" \\ status = 500`
- [ ] `http "http://localhost:5959" \\ status = 404`

### CODE REVIEW

File: word-api/lib/words.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+				});
+
+				return resolve(text);
+			}
+		});
+	});
+}
+
+/**
+ * Count words in a string
```
Comment (fluxsauce) [inline word-api/lib/words.js]: The jsdoc description is not quite accurate; it scrapes a remote url then returns a word count from the contents.

File: word-api/lib/words.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+/**
+ * Get html from url
+ *
+ * @param  {string} url - url of html to scrape.
+ *
+ * @return {string} - text from html.
+ */
+function getText(url) {
+	return new Promise(function (resolve, reject) {
+		let count = 0;
```
Comment (fluxsauce) [inline word-api/lib/words.js]: Never used?

**Review by fluxsauce — CHANGES_REQUESTED** [review]
Good progress so far!

The `http` client has some shortcuts. Check out https://github.com/jakubroztocil/httpie for examples

- `http :5959/ping`
  - Returns PONG
- `http :5959/api/words url==http://www.kink.com/shoot/39563`
  - Good, it handles redirects to https
- `http :5959/api/words url==https://gist.githubusercontent.com/fluxsauce/816c1aca8ddb3cc2783b87ff8b12114a/raw/870cc51e1dd710314608711cbcf95a0c39548aac/gistfile1.txt`
  - Good, returns 10 as I expected from a raw text file
- `http :5959/api/words url==https://gist.githubusercontent.com/fluxsauce/a8b98c390ce9a824556f1f7382e291c7/raw/d04f91b22b91b26044833936a47d77bcd2d04609/test.html`
  - Good, returns 7 as I expected from arbitrary HTML
- `http :5959/api/words`
  - Returns a 500, which is a server error. What HTTP status code should you use for client errors?
- `http :5959`
  - Good, returns 404
- `http :5959/api/words url==http://www.slate.com/content/dam/slate/articles/health_and_science/science/2017/12/171213_SCI_Oumuamua-Aliens.jpg.CROP.promo-xlarge2.jpg`
  - Bad, returned 10731 - it's an image

Things that would make it better:

- Request logging - log all requests prior to execution with `console.log` in a middleware
- Log errors when they occur with `console.error`
- Check if input is binary or text
- Use correct HTTP status
- `/node_modules/.bin/eslint .`
- Use a `const` for immutable values

File: word-api/lib/words.js (hunk 2 above)
Comment (tavizon) [reply word-api/lib/words.js]: Correct. Removed as it was for functionality moved to the count method.

File: word-api/lib/words.js (hunk 1 above)
Comment (tavizon) [reply word-api/lib/words.js]: Updated.

**Comment by tavizon** [comment]
I believe all feedback has been addressed :crossed_fingers:

**Review by fluxsauce — APPROVED** [review]
Looks good! Result:

```
16|app     | [2017-12-15 14:31:35.316] [INFO] [CodingChallenge/word-api/app.js] [listen] [NOREQID] word-api - listening on port 5959
16|app     | [2017-12-15 14:31:51.457] [INFO] [CodingChallenge/word-api/routes/index.js] [NOACTION] [NOREQID] {"url":"http://www.kink.com/shoot/39563"}
16|app     | [2017-12-15 14:32:00.855] [INFO] [CodingChallenge/word-api/routes/index.js] [NOACTION] [NOREQID] {"url":"https://gist.githubusercontent.com/fluxsauce/816c1aca8ddb3cc2783b87ff8b12114a/raw/870cc51e1dd710314608711cbcf95a0c39548aac/gistfile1.txt"}
16|app     | [2017-12-15 14:32:05.043] [INFO] [CodingChallenge/word-api/routes/index.js] [NOACTION] [NOREQID] {"url":"https://gist.githubusercontent.com/fluxsauce/a8b98c390ce9a824556f1f7382e291c7/raw/d04f91b22b91b26044833936a47d77bcd2d04609/test.html"}
16|app     | [2017-12-15 14:32:07.394] [INFO] [CodingChallenge/word-api/routes/index.js] [NOACTION] [NOREQID] {}
16|app     | [2017-12-15 14:32:11.816] [INFO] [CodingChallenge/word-api/routes/index.js] [NOACTION] [NOREQID] {}
16|app     | [2017-12-15 14:32:14.541] [INFO] [CodingChallenge/word-api/routes/index.js] [NOACTION] [NOREQID] {"url":"http://www.slate.com/content/dam/slate/articles/health_and_science/science/2017/12/171213_SCI_Oumuamua-Aliens.jpg.CROP.promo-xlarge2.jpg"}
16|app     | [2017-12-15 14:32:14.753] [ERROR] [CodingChallenge/word-api/routes/api/words.js] [NOACTION] [NOREQID] URL contains no text
```

**Pull request merged**